templates/observables/br_d_l_to_d_l.cpp.in

```
@@ -0,0 +1,267 @@
+ *       Minus comes from the form_factors embedding into four-fermion
+ *       amplitude, because we use descending ordering for external fermions
+ *          <out:l4,q3| T exp(i L_@ModelName@ dx) |in:l2,q1> =: G4
+ * @note For four fermion coefficient matching minus comes from the G4
+ *       definition, because
+ *          <out:l4,q3| T exp(i L_low dx) |in:l2,q1> = -i * C_XY [3X1]*[4Y2].
+ * @tparam    Lepton   Type of a lepton fields.
+ * @tparam    Photon   Type of a photon field.
+ * @tparam    photon   Function name for photon form factors.
+ * @tparam    npf      Function name for npf function for up-quark contribution.
+ * @param[in] dIn     Generation index for incoming quark.
+ * @param[in] dOut    Generation index for outgoing quark.
+ * @param[in] lInOut  Generation index for leptons.
+ * @param[in] model  Mass eigenstates.
+ * @param[in] parameters Parameters for the observable calculation.
+ * @param[in] qedqcd Reference to low-energy data.
+ * @return A dummy zero observable and Wilson coefficients.
+ */
+template<class Down, class Lepton, class Photon, ffv_function photon, npf_function npf>
+Eigen::Array<std::complex<double>, 13, 1> forge(
+   int dIn, int dOut, int lInOut,
+   const @ModelName@_mass_eigenstates& model,
+   const softsusy::QedQcd& qedqcd) {
+
+   context_base context {model};
+
+   const auto form_factors = photon(dIn, dOut, model, true);
+   const auto photon_l = embed_photon<Lepton, Photon>(model, form_factors, lInOut);
+   const auto npf_l = fix_tensors_sign<npf>(model, dIn, dOut, lInOut);
+
+   // Matching
+   const auto DL = - 0.5 * form_factors[2];
+   const auto DR = - 0.5 * form_factors[3];
+   const auto CXY = - photon_l - npf_l;
+
+   if (dIn == 2 && dOut == 1 && lInOut == 1) {
+      write_btosmumu<Down>(CXY, context, qedqcd);
+   } else {
+      // TODO(all): Implement :)
+      ERROR("dltodl: Unimplemented combination of external particles.");
+      exit(EXIT_FAILURE);
```

See comment above, please throw an exception.